packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/workflow-actions/hooks/useDiscardDraftWorkflowSingleRecordAction.ts

```diff
@@ -0,0 +1,63 @@
+import { useActionMenuEntries } from '@/action-menu/hooks/useActionMenuEntries';
+import {
+  ActionMenuEntryScope,
+  ActionMenuEntryType,
+} from '@/action-menu/types/ActionMenuEntry';
+import { useDeleteOneWorkflowVersion } from '@/workflow/hooks/useDeleteOneWorkflowVersion';
+import { useWorkflowWithCurrentVersion } from '@/workflow/hooks/useWorkflowWithCurrentVersion';
+import { IconTrash, isDefined } from 'twenty-ui';
+
+export const useDiscardDraftWorkflowSingleRecordAction = ({
+  workflowId,
+}: {
+  workflowId: string;
+}) => {
+  const { addActionMenuEntry, removeActionMenuEntry } = useActionMenuEntries();
+
+  const { deleteOneWorkflowVersion } = useDeleteOneWorkflowVersion();
+
+  const workflowWithCurrentVersion = useWorkflowWithCurrentVersion(workflowId);
+
+  const registerDiscardDraftWorkflowSingleRecordAction = ({
+    position,
+  }: {
+    position: number;
+  }) => {
+    if (
+      !isDefined(workflowWithCurrentVersion) ||
+      !isDefined(workflowWithCurrentVersion.currentVersion.trigger)
+    ) {
+      return;
+    }
+
+    const isDraft =
+      workflowWithCurrentVersion.currentVersion.status === 'DRAFT';
+
+    if (!isDraft) {
```

Comment:
The number of workflowVersions should be strictly greater that 1 to be able to perform `DiscardDraft` operation. You should add that condition here